ntoskrnl/inbv/inbv.c

```diff
@@ -932,18 +948,67 @@ DisplayBootBitmap(IN BOOLEAN TextMode)
     }
 
 #ifdef INBV_ROTBAR_IMPLEMENTED
+    /* Hide simple progress bar if not used */
+    if (RotBarSelection != RB_PROGRESS_BAR)
+        ShowProgressBar = FALSE;
+
     /* Do we have a system thread? */
     if (RotBarThreadActive)
     {
         /* We do, initialize the progress bar */
         InbvAcquireLock();
         RotBarSelection = TempRotBarSelection;
-        // InbvRotBarInit();
+        InbvRotBarInit();
         InbvReleaseLock();
     }
 #endif
 }
 
+VOID
+NTAPI
+INIT_FUNCTION
+InbvRotBarInit(VOID)
+{
+    ULONG X, Y, Iteration, Index, Total = 18;
+    UCHAR Buffer[128];
+#ifndef INBV_ROTBAR_IMPLEMENTED
```

Comment:
Instead of this you could do:
```
#ifdef INBV_ROTBAR_IMPLEMENTED
VOID NTAPI ... InbvRotBarInit(VOID)
{
    // The implementation
    return;
}
#else
VOID NTAPI ... InbvRotBarInit(VOID)
{
    return;
}
#endif
```
or even, just have the function defined in that case and calling in the main code only between `#ifdef INBV_ROTBAR_IMPLEMENTED` guards ?